File: Homeworks/01.CSharp-Basic/04.Console-Input-Output/05.FormattingNumbers/FormattingNumbers.cs
using System;
using System.Globalization;

/* Formatting Numbers
 *  Write a program that reads 3 numbers: an integer a (0 ≤ a ≤ 500),
 *  a floating-point b and a floating-point c and prints them in 4 virtual columns on the console.
 *  Each column should have a width of 10 characters. The number a should be printed in hexadecimal, left aligned;
 *  then the number a should be printed in binary form, padded with zeroes,
 *  then the number b should be printed with 2 digits after the decimal point, right aligned;
 *  the number c should be printed with 3 digits after the decimal point, left aligned.
 */

class FormattingNumbers
{
    static void Main()
    {
        #region User Input
        Console.Write("Enter number a: ");
        ushort number = ushort.Parse(Console.ReadLine());

        Console.Write("Enter number b: ");
        string firstFloatInput = Console.ReadLine().Replace(',', '.');
        float firstFloatNumber = float.Parse(firstFloatInput, CultureInfo.InvariantCulture);

        Console.Write("Enter number c: ");
        string secondFloatInput = Console.ReadLine().Replace(',', '.');
        float secondFloatNumber = float.Parse(secondFloatInput, CultureInfo.InvariantCulture);
        #endregion

        Console.WriteLine("|{0,-10}|{1}|{2,10:0.00}|{3,-10:0.000}|",
            number.ToString("X"), Convert.ToString(number, 2).PadLeft(10, '0'), firstFloatNumber, secondFloatNumber);
    }
}

File: Homeworks/02.Java/02.Java-Syntax/src/FormattingNumbers.java
import java.util.Scanner;

// 6. Write a program that reads 3 numbers: an integer a (0 ≤ a ≤ 500),
//	  a floating-point b and a floating-point c and prints them in 4 virtual columns on the console.
//	  Each column should have a width of 10 characters. The number a should be printed in hexadecimal,
//	  left aligned; then the number a should be printed in binary form,
//	  padded with zeroes, then the number b should be printed with 2 digits after the decimal point, right aligned;
//	  the number c should be printed with 3 digits after the decimal point, left aligned.

public class FormattingNumbers {

	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);

		System.out.print("Number a: ");
		int a = in.nextInt();

		System.out.print("Number b: ");
		double b = in.nextDouble();

		System.out.print("Number c: ");
		double c = in.nextDouble();
		
		in.close();

		String hexString = Integer.toHexString(a).toUpperCase();
		String binString = Integer.toBinaryString(a);

		System.out.printf("\n|%-10s|%010d|%10.2f|%-10.3f|", hexString,
				Integer.parseInt(binString), b, c);
	}
}